velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1096,6 +1096,31 @@ struct ParseDateTimeFunction {
   }
 };
 
+template <typename T>
+struct CurrentTimeStampFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  std::optional<int64_t> sessionTzID_;
+
+  FOLLY_ALWAYS_INLINE void initialize(const core::QueryConfig& config) {
+    auto sessionTzName = config.sessionTimezone();
+    if (!sessionTzName.empty()) {
+      sessionTzID_ = util::getTimeZoneID(sessionTzName);
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void call(out_type<TimestampWithTimezone>& result) {
+    int64_t current_timestamp = Timestamp::now().toMillis();
+
+    std::optional<int64_t> sessionTzID_;
+    // Set the timezone to the session timezone. If there's
+    // no session timezone, fallback to 0 (GMT).
+    int16_t timezoneId = sessionTzID_.value_or(0);
```

Comment:
This timezoneId can be computed in the initialize function itself. It needn't be re-computed for each row call.